### PR TITLE
Add JMX connector implementation to framework

### PR DESCRIPTION
**Pull request by BenNzewi**

### CODE REVIEW

File: jmx/src/test/java/uk/gov/justice/services/jmx/MBeanFactoryTest.java [hunk 1]
```diff
@@ -21,10 +20,11 @@
 public class MBeanFactoryTest {
 
     @Mock
-    private DefaultShutteringMBean defaultShutteringMBean;
+    private Shuttering shuttering
+            ;
```
Comment (allanmckenzie) [inline jmx/src/test/java/uk/gov/justice/services/jmx/MBeanFactoryTest.java]: Is this a rogue extra line of whitespace?

File: jmx/src/main/java/uk/gov/justice/services/jmx/MBeanHelper.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        final MBeanInfo mBeanInfo = connection.getMBeanInfo(objectName);
+        MBeanOperationInfo[] operations = mBeanInfo.getOperations();
+        List<MBeanOperationInfo> mbeanOperations = Arrays.asList(operations);
+
+        LOGGER.info("MBean Operations: ");
+        mbeanOperations.forEach(mBeanOperationInfo -> LOGGER.info(mBeanOperationInfo.getName()));
+    }
+
+    public JMXConnector getJMXConnector() throws IOException {
+        int managementPort = Integer.valueOf(System.getProperty(RANDOM_MANAGEMENT_PORT));
```
Comment (mapingo) [inline jmx/src/main/java/uk/gov/justice/services/jmx/MBeanHelper.java]: final ?

File: jmx/src/main/java/uk/gov/justice/services/jmx/MBeanHelper.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+        String urlString =
```
Comment (mapingo) [inline jmx/src/main/java/uk/gov/justice/services/jmx/MBeanHelper.java]: final

File: jmx/src/main/java/uk/gov/justice/services/jmx/MBeanHelper.java [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+                System.getProperty("jmx.service.url","service:jmx:remote+http://" + HOST + ":" + managementPort);
+        JMXServiceURL serviceURL = new JMXServiceURL(urlString);
```
Comment (mapingo) [inline jmx/src/main/java/uk/gov/justice/services/jmx/MBeanHelper.java]: final

File: jmx/src/main/java/uk/gov/justice/services/jmx/MBeanHelper.java [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+import javax.management.remote.JMXConnector;
+import javax.management.remote.JMXConnectorFactory;
+import javax.management.remote.JMXServiceURL;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class MBeanHelper {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(MBeanHelper.class.getName());
```
Comment (allanmckenzie) [inline jmx/src/main/java/uk/gov/justice/services/jmx/MBeanHelper.java]: Inject the Logger

File: jmx/src/main/java/uk/gov/justice/services/jmx/MBeanHelper.java [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+
+    private static final String HOST = "localhost";
+    private static final String RANDOM_MANAGEMENT_PORT = "random.management.port";
+
+    public ShutteringMBean getMbeanProxy(ObjectName objectName, final MBeanServerConnection connection) {
```
Comment (allanmckenzie) [inline jmx/src/main/java/uk/gov/justice/services/jmx/MBeanHelper.java]: final :-)

**Pull request merged**